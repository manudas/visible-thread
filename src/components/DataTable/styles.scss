.fixedTable {
    width: 250px;
    height: fit-content;
}

.table tr:nth-child(even) {
    background: #e4e4e4;
}