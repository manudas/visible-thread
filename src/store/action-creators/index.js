import {
    fetchTeamsScans,
} from '../../services';

export const FETCH_TEAMS_SCANS_TYPE = 'FETCH_TEAMS_SCANS';
export const FETCH_TEAMS_SCANS = (periods, division) => {
    return async (dispatch) => {

        const teamsData = await fetchTeamsScans(periods, division);

        dispatch( {
            type: FETCH_TEAMS_SCANS_TYPE,
            payload: teamsData
        });
    }
};
