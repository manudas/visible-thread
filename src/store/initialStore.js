import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import {
    composeWithDevTools,
} from 'redux-devtools-extension';

import rootReducer from './reducers';

// no need of Thunk middleware as is not really asynchronous data fetching
export default createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
