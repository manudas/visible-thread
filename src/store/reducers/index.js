import { combineReducers } from 'redux';

import {
    fetchTeamsReducer,
} from './teams';

/**
 * The root reducer of the app
 */
export default combineReducers({
    teams: fetchTeamsReducer,
});
