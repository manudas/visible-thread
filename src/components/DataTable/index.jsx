import { useEffect, useState } from 'react';
import {
    useSelector,
} from 'react-redux';

import Popup from '../Popup';

import { fetchMonth } from '../../services'

import './styles.scss';

const DataTable = () => {
    const [data, setData] = useState({});
    const [weeklyData, setWeeklyData] = useState(null);
    const teamData = useSelector((state) => {
        return state.teams ?? {};
    });

    const buildDataTable = (data) => {
        let result = null;
        if (data) {
            const groupedData = data.reduce((carry, {
                teamName,
                date,
                scansAMonth,
                scansAWeek,
                totalScans,
            }) => {
                if(!carry[teamName]) carry[teamName] = {data: []};
                if(!carry.headers) carry.headers = [];
                if(!totalScans) {
                    const _date = new Date(date);
                    const month = _date.getUTCMonth();
                    const monthText = _date.toLocaleString('default', { month: 'short' });
                    const year = _date.getUTCFullYear();
                    const dayOfWeek = _date.getUTCDay();
                    const dayOfMonth = _date.getUTCDate();
                    const weekOfMonth = Math.ceil((dayOfMonth - 1 - dayOfWeek) / 7);

                    if(!carry[teamName].data[year]) carry[teamName].data[year] = [];

                    if (scansAWeek !== undefined && !carry[teamName].data[year][month]) carry[teamName].data[year][month] = [];

                    if(!carry.headers[year]) {
                        carry.headers[year] = [];
                    }
                    if (!carry.headers[year][month]) {

                        carry.headers[year][month] = scansAWeek !== undefined ? [] : monthText;

                    }
                    if (scansAWeek !== undefined) {
                        if (!carry.headers[year][month].includes(weekOfMonth)) {
                            carry.headers[year][month].push(weekOfMonth);
                        }
                        carry[teamName].data[year][month].push(scansAWeek);
                    } else {
                        carry[teamName].data[year][month] = scansAMonth;
                    }
                }
                else {
                    const elementCounter = Object.keys(carry[teamName]).length;
                    // response is ordered so totalScan is the last key
                    carry[teamName].totalScans = totalScans;
                    carry[teamName].average = totalScans/elementCounter;
                }
                return carry;
            }, {});

            result = groupedData;
        }

        return result;
    };

    useEffect(() => {
        const preparedData = buildDataTable(teamData);
        setData(preparedData);
    }, [teamData]);

    return (
        Object.keys(data).length > 0
        ?
            <section className="data w-100 text-nowrap d-flex">
                <table className="table fixedTable">
                    <thead>
                        <tr>
                            <th>Team</th>
                            <th>Average</th>
                            <th>Total</th>
                        </tr>
                    </thead>
                    <tbody>
                        {
                            Object.keys(data).map(key => {
                                if (key === 'headers') return null;
                                const {
                                    average,
                                    totalScans
                                } = data[key];

                                return  (
                                    <tr
                                        key={key}
                                    >
                                        <td>{key}</td>
                                        <td>{Math.round(average * 100) / 100}</td>
                                        <td>{totalScans}</td>
                                    </tr>
                                );
                            })
                        }
                    </tbody>
                </table>
                <div className="w-75 overflow-auto">
                    <table className="table" data-testid="breakdown-results-table">
                        <thead>
                            <tr>
                                {
                                    data.headers.map((element, year) => {
                                        return element.map((month, indexMonth) => {
                                            return !Array.isArray(month)
                                                ?
                                                    <th
                                                        key={`${indexMonth}-${month}`}
                                                    >
                                                        {month}
                                                    </th>
                                                : month.map((week, indexWeek) => {
                                                    return <th
                                                        key={`${indexMonth}-${month}-${indexWeek}`}
                                                    >
                                                        Month: {indexMonth}, week: {week}
                                                    </th>
                                                })
                                        })
                                    })
                                }
                            </tr>
                        </thead>
                        <tbody>
                            {
                                Object.keys(data).map(key => {
                                    if (key === 'headers') return null;
                                    const {
                                        data: teamData
                                    } = data[key];

                                    return <tr
                                        key={key}
                                    >
                                        {
                                            teamData.map((element, year) =>
                                                element.map((monthTotal, monthIndex) => {
                                                    // if(!monthTotal) console.log(monthIndex, monthTotal);
                                                    return !Array.isArray(monthTotal)
                                                        ?
                                                            <td
                                                                key={monthIndex}
                                                                onClick={async() => {
                                                                    if (monthTotal) {
                                                                        const breakDown = await fetchMonth(key, `${year}-${monthIndex + 1}-01`);
                                                                        setWeeklyData(breakDown);
                                                                    }
                                                                }}
                                                            >
                                                                {monthIndex}
                                                            </td>
                                                        : monthTotal.map((weekTotal, indexWeek) => {
                                                            return <td
                                                                key={`${monthIndex}-${indexWeek}`}
                                                            >
                                                                {weekTotal}
                                                            </td>
                                                        })
                                                })
                                            )
                                        }
                                    </tr>
                                })
                            }
                        </tbody>
                    </table>
                </div>
                <Popup data={weeklyData} closePopup={() => setWeeklyData(null)} />
            </section>
        : null
    )
}

export default DataTable;
