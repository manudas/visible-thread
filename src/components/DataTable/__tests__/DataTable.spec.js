import React from 'react';
import {
    fireEvent,
    render,
    waitFor,
} from '@testing-library/react';

import { Provider } from 'react-redux'
import store from '../../../store/initialStore';

import DataTable from '..';
import monthData from '../../../__mockedData__/teams-month.json';
import weekData from '../../../__mockedData__/teams-week.json';
import monthBreakdownData from '../../../__mockedData__/teams-month-week-breakdown.json';
import { FETCH_TEAMS_SCANS } from '../../../store/action-creators';

const ReduxProvider = ({ children, reduxStore }) => (
    <Provider store={reduxStore}>{children}</Provider>
);

describe('DataTable test suite', () => {

    it('should render emtpy <DataTable />', () => {
        const renderer = render(
            <ReduxProvider reduxStore={store}>
                <DataTable />
            </ReduxProvider>
        );
        const { container } = renderer;
        expect(container).toMatchSnapshot();
    });

    it('should render <DataTable /> with monthly scans', async () => {
        jest.spyOn(window, 'fetch').mockImplementation(async () => {
            return {
                ok: true,
                json: async () => monthData,
            };
        });
        store.dispatch(FETCH_TEAMS_SCANS(12, 'months'));
        const renderer = render(
            <ReduxProvider reduxStore={store}>
                <DataTable />
            </ReduxProvider>
        );
        const {
            container,
            getByTestId,
        } = renderer;

        // wait till the table is filled in with the data in the store
        await waitFor(() => {
            expect(getByTestId('breakdown-results-table')).toBeInTheDocument();
        });
        expect(container).toMatchSnapshot();
    });

    it('should render <DataTable /> with weekly scans', async () => {
        jest.spyOn(window, 'fetch').mockImplementation(async () => {
            return {
                ok: true,
                json: async () => weekData,
            };
        });
        store.dispatch(FETCH_TEAMS_SCANS(12, 'weeks'));
        const renderer = render(
            <ReduxProvider reduxStore={store}>
                <DataTable />
            </ReduxProvider>
        );
        const {
            container,
            getByTestId,
        } = renderer;

        // wait till the table is filled in with the data in the store
        await waitFor(() => {
            expect(getByTestId('breakdown-results-table')).toBeInTheDocument();
        });
        expect(container).toMatchSnapshot();
    });

    it('should render weekly breakdown data in <DataTable /> for monthly scans', async() => {
        jest.spyOn(window, 'fetch').mockImplementation(async () => {
            return {
                ok: true,
                json: async () => monthData,
            };
        });
        store.dispatch(FETCH_TEAMS_SCANS(12, 'months'));

        const renderer = render(
            <ReduxProvider reduxStore={store}>
                <DataTable />
            </ReduxProvider>
        );

        const {
            getByTestId,
            container
        } = renderer;


        // wait till the table is filled in with the data in the store
        await waitFor(() => {
            expect(getByTestId('breakdown-results-table')).toBeInTheDocument();
        });
        const endpointSpyFunc = jest.spyOn(window, 'fetch').mockImplementation(async () => {
            return {
                ok: true,
                json: async () => monthBreakdownData,
            };
        });

        const table = getByTestId('breakdown-results-table');

        const cells = table.getElementsByTagName('td');
        fireEvent.click(
            cells[0], // first cell
        );

        expect(endpointSpyFunc).toHaveBeenCalled();

        await waitFor(() => {
            expect(getByTestId('popup__container')).toBeInTheDocument();
        });

        expect(container).toMatchSnapshot();
    });
});
