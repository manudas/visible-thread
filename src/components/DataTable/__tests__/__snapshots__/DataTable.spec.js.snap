// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable test suite should render <DataTable /> with monthly scans 1`] = `
<div>
  <section
    class="data w-100 text-nowrap d-flex"
  >
    <table
      class="table fixedTable"
    >
      <thead>
        <tr>
          <th>
            Team
          </th>
          <th>
            Average
          </th>
          <th>
            Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Team 1
          </td>
          <td>
            4958
          </td>
          <td>
            4958
          </td>
        </tr>
        <tr>
          <td>
            Team 2
          </td>
          <td>
            6986
          </td>
          <td>
            6986
          </td>
        </tr>
        <tr>
          <td>
            Team 3
          </td>
          <td>
            7265
          </td>
          <td>
            7265
          </td>
        </tr>
        <tr>
          <td>
            Team 4
          </td>
          <td>
            6527
          </td>
          <td>
            6527
          </td>
        </tr>
        <tr>
          <td>
            Team 5
          </td>
          <td>
            4995
          </td>
          <td>
            4995
          </td>
        </tr>
        <tr>
          <td>
            Team 6
          </td>
          <td>
            6383
          </td>
          <td>
            6383
          </td>
        </tr>
        <tr>
          <td>
            Team 7
          </td>
          <td>
            4904
          </td>
          <td>
            4904
          </td>
        </tr>
        <tr>
          <td>
            Team 8
          </td>
          <td>
            8003
          </td>
          <td>
            8003
          </td>
        </tr>
        <tr>
          <td>
            Team 9
          </td>
          <td>
            4595
          </td>
          <td>
            4595
          </td>
        </tr>
        <tr>
          <td>
            Team 10
          </td>
          <td>
            5260
          </td>
          <td>
            5260
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="w-75 overflow-auto"
    >
      <table
        class="table"
        data-testid="breakdown-results-table"
      >
        <thead>
          <tr>
            <th>
              Oct
            </th>
            <th>
              Nov
            </th>
            <th>
              Dec
            </th>
            <th>
              Jan
            </th>
            <th>
              Feb
            </th>
            <th>
              Mar
            </th>
            <th>
              Apr
            </th>
            <th>
              May
            </th>
            <th>
              Jun
            </th>
            <th>
              Jul
            </th>
            <th>
              Aug
            </th>
            <th>
              Sep
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</div>
`;

exports[`DataTable test suite should render <DataTable /> with weekly scans 1`] = `
<div>
  <section
    class="data w-100 text-nowrap d-flex"
  >
    <table
      class="table fixedTable"
    >
      <thead>
        <tr>
          <th>
            Team
          </th>
          <th>
            Average
          </th>
          <th>
            Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Team 1
          </td>
          <td>
            1248
          </td>
          <td>
            1248
          </td>
        </tr>
        <tr>
          <td>
            Team 2
          </td>
          <td>
            1655
          </td>
          <td>
            1655
          </td>
        </tr>
        <tr>
          <td>
            Team 3
          </td>
          <td>
            1995
          </td>
          <td>
            1995
          </td>
        </tr>
        <tr>
          <td>
            Team 4
          </td>
          <td>
            1630
          </td>
          <td>
            1630
          </td>
        </tr>
        <tr>
          <td>
            Team 5
          </td>
          <td>
            1529
          </td>
          <td>
            1529
          </td>
        </tr>
        <tr>
          <td>
            Team 6
          </td>
          <td>
            1469
          </td>
          <td>
            1469
          </td>
        </tr>
        <tr>
          <td>
            Team 7
          </td>
          <td>
            1575
          </td>
          <td>
            1575
          </td>
        </tr>
        <tr>
          <td>
            Team 8
          </td>
          <td>
            1389
          </td>
          <td>
            1389
          </td>
        </tr>
        <tr>
          <td>
            Team 9
          </td>
          <td>
            2125
          </td>
          <td>
            2125
          </td>
        </tr>
        <tr>
          <td>
            Team 10
          </td>
          <td>
            1340
          </td>
          <td>
            1340
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="w-75 overflow-auto"
    >
      <table
        class="table"
        data-testid="breakdown-results-table"
      >
        <thead>
          <tr>
            <th>
              Month: 
              6
              , week: 
              4
            </th>
            <th>
              Month: 
              6
              , week: 
              3
            </th>
            <th>
              Month: 
              6
              , week: 
              2
            </th>
            <th>
              Month: 
              7
              , week: 
              3
            </th>
            <th>
              Month: 
              7
              , week: 
              2
            </th>
            <th>
              Month: 
              7
              , week: 
              1
            </th>
            <th>
              Month: 
              7
              , week: 
              0
            </th>
            <th>
              Month: 
              8
              , week: 
              3
            </th>
            <th>
              Month: 
              8
              , week: 
              2
            </th>
            <th>
              Month: 
              8
              , week: 
              1
            </th>
            <th>
              Month: 
              8
              , week: 
              0
            </th>
            <th>
              Month: 
              9
              , week: 
              0
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              208
            </td>
            <td>
              112
            </td>
            <td>
              82
            </td>
            <td>
              20
            </td>
            <td>
              132
            </td>
            <td>
              94
            </td>
            <td>
              173
            </td>
            <td>
              79
            </td>
            <td>
              109
            </td>
            <td>
              147
            </td>
            <td>
              77
            </td>
            <td>
              15
            </td>
          </tr>
          <tr>
            <td>
              214
            </td>
            <td>
              114
            </td>
            <td>
              163
            </td>
            <td>
              3
            </td>
            <td>
              116
            </td>
            <td>
              198
            </td>
            <td>
              247
            </td>
            <td>
              232
            </td>
            <td>
              141
            </td>
            <td>
              207
            </td>
            <td>
              5
            </td>
            <td>
              15
            </td>
          </tr>
          <tr>
            <td>
              245
            </td>
            <td>
              162
            </td>
            <td>
              143
            </td>
            <td>
              203
            </td>
            <td>
              227
            </td>
            <td>
              14
            </td>
            <td>
              52
            </td>
            <td>
              176
            </td>
            <td>
              231
            </td>
            <td>
              167
            </td>
            <td>
              219
            </td>
            <td>
              156
            </td>
          </tr>
          <tr>
            <td>
              31
            </td>
            <td>
              115
            </td>
            <td>
              188
            </td>
            <td>
              1
            </td>
            <td>
              137
            </td>
            <td>
              142
            </td>
            <td>
              93
            </td>
            <td>
              133
            </td>
            <td>
              218
            </td>
            <td>
              166
            </td>
            <td>
              205
            </td>
            <td>
              201
            </td>
          </tr>
          <tr>
            <td>
              128
            </td>
            <td>
              162
            </td>
            <td>
              15
            </td>
            <td>
              92
            </td>
            <td>
              13
            </td>
            <td>
              218
            </td>
            <td>
              244
            </td>
            <td>
              160
            </td>
            <td>
              242
            </td>
            <td>
              135
            </td>
            <td>
              24
            </td>
            <td>
              96
            </td>
          </tr>
          <tr>
            <td>
              80
            </td>
            <td>
              190
            </td>
            <td>
              159
            </td>
            <td>
              204
            </td>
            <td>
              217
            </td>
            <td>
              23
            </td>
            <td>
              71
            </td>
            <td>
              8
            </td>
            <td>
              22
            </td>
            <td>
              128
            </td>
            <td>
              167
            </td>
            <td>
              200
            </td>
          </tr>
          <tr>
            <td>
              138
            </td>
            <td>
              14
            </td>
            <td>
              215
            </td>
            <td>
              191
            </td>
            <td>
              97
            </td>
            <td>
              31
            </td>
            <td>
              136
            </td>
            <td>
              188
            </td>
            <td>
              209
            </td>
            <td>
              233
            </td>
            <td>
              122
            </td>
            <td>
              1
            </td>
          </tr>
          <tr>
            <td>
              49
            </td>
            <td>
              57
            </td>
            <td>
              68
            </td>
            <td>
              24
            </td>
            <td>
              210
            </td>
            <td>
              134
            </td>
            <td>
              108
            </td>
            <td>
              112
            </td>
            <td>
              244
            </td>
            <td>
              160
            </td>
            <td>
              105
            </td>
            <td>
              118
            </td>
          </tr>
          <tr>
            <td>
              232
            </td>
            <td>
              153
            </td>
            <td>
              248
            </td>
            <td>
              90
            </td>
            <td>
              207
            </td>
            <td>
              231
            </td>
            <td>
              169
            </td>
            <td>
              171
            </td>
            <td>
              95
            </td>
            <td>
              236
            </td>
            <td>
              197
            </td>
            <td>
              96
            </td>
          </tr>
          <tr>
            <td>
              60
            </td>
            <td>
              143
            </td>
            <td>
              137
            </td>
            <td>
              104
            </td>
            <td>
              41
            </td>
            <td>
              205
            </td>
            <td>
              59
            </td>
            <td>
              109
            </td>
            <td>
              80
            </td>
            <td>
              84
            </td>
            <td>
              161
            </td>
            <td>
              157
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</div>
`;

exports[`DataTable test suite should render emtpy <DataTable /> 1`] = `<div />`;

exports[`DataTable test suite should render weekly breakdown data in <DataTable /> for monthly scans 1`] = `
<div>
  <section
    class="data w-100 text-nowrap d-flex"
  >
    <table
      class="table fixedTable"
    >
      <thead>
        <tr>
          <th>
            Team
          </th>
          <th>
            Average
          </th>
          <th>
            Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Team 1
          </td>
          <td>
            4958
          </td>
          <td>
            4958
          </td>
        </tr>
        <tr>
          <td>
            Team 2
          </td>
          <td>
            6986
          </td>
          <td>
            6986
          </td>
        </tr>
        <tr>
          <td>
            Team 3
          </td>
          <td>
            7265
          </td>
          <td>
            7265
          </td>
        </tr>
        <tr>
          <td>
            Team 4
          </td>
          <td>
            6527
          </td>
          <td>
            6527
          </td>
        </tr>
        <tr>
          <td>
            Team 5
          </td>
          <td>
            4995
          </td>
          <td>
            4995
          </td>
        </tr>
        <tr>
          <td>
            Team 6
          </td>
          <td>
            6383
          </td>
          <td>
            6383
          </td>
        </tr>
        <tr>
          <td>
            Team 7
          </td>
          <td>
            4904
          </td>
          <td>
            4904
          </td>
        </tr>
        <tr>
          <td>
            Team 8
          </td>
          <td>
            8003
          </td>
          <td>
            8003
          </td>
        </tr>
        <tr>
          <td>
            Team 9
          </td>
          <td>
            4595
          </td>
          <td>
            4595
          </td>
        </tr>
        <tr>
          <td>
            Team 10
          </td>
          <td>
            5260
          </td>
          <td>
            5260
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="w-75 overflow-auto"
    >
      <table
        class="table"
        data-testid="breakdown-results-table"
      >
        <thead>
          <tr>
            <th>
              Oct
            </th>
            <th>
              Nov
            </th>
            <th>
              Dec
            </th>
            <th>
              Jan
            </th>
            <th>
              Feb
            </th>
            <th>
              Mar
            </th>
            <th>
              Apr
            </th>
            <th>
              May
            </th>
            <th>
              Jun
            </th>
            <th>
              Jul
            </th>
            <th>
              Aug
            </th>
            <th>
              Sep
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              10
            </td>
            <td>
              11
            </td>
            <td>
              0
            </td>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              7
            </td>
            <td>
              8
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="popup__container"
      data-testid="popup__container"
    >
      <div
        class="popup__content bg-light"
      >
        <div>
          Total scans in week 
          0
          : 
          89
        </div>
        <div>
          Total scans in week 
          1
          : 
          70
        </div>
        <div>
          Total scans in week 
          2
          : 
          164
        </div>
        <div>
          Total scans in week 
          3
          : 
          200
        </div>
        <div>
          Total scans in week 
          4
          : 
          179
        </div>
      </div>
    </div>
  </section>
</div>
`;
